File: kinotic-domain/src/main/java/org/kinotic/os/api/model/Project.java
package org.kinotic.os.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
public class Project implements ApplicationScoped<String> {

    /**
     * The id of the project.
     * All project ids are unique throughout the entire system.
     */
    private String id;

    private String organizationId;

    /**
     * The id of the application that this project belongs to.
     * All application ids are unique throughout the entire system.
     */
    private String applicationId;

    /**
     * The name of the project.
     */
    private String name;

    /**
     * The description of the project.
     */
    private String description;

    /**
     * The source of truth for the project.
     */
    private ProjectType sourceOfTruth;

    /**
     * Full name ({@code owner/repo}) of the GitHub repository backing this project.
     * Stamped at create time by the repo provisioner; not synced back from GitHub.
     */
    private String repoFullName;

    /**
     * GitHub's stable repository id. Survives rename on the GitHub side.
     */
    private Long repoId;

    /**
     * Default branch of the backing repository at the time it was provisioned
     * (e.g. {@code main}).
     */
    private String defaultBranch;

    /**
     * Visibility chosen for the backing repository at create time. The SPA sets
     * this before save; the platform passes it through to GitHub.
     */
    private boolean repoPrivate;

    /**
     * Connection state between this project and its backing GitHub repository.
     * {@link RepositoryConnectionStatus#CONNECTED} at provision time;
     * webhook handlers flip to {@link RepositoryConnectionStatus#DISCONNECTED}
     * when GitHub revokes access to the repo.
     */
    private RepositoryConnectionStatus repositoryConnectionStatus;

    /**
     * The date and time the project was updated.
     */
    private Date updated;

}

File: kinotic-js/workspace/packages/os-api/src/api/model/Project.ts
import type { Identifiable } from '@kinotic-ai/core'
import { ProjectType } from '@/api/model/ProjectType'

export class Project implements Identifiable<string> {
    /**
     * The id of the project.
     * All project ids are unique throughout the entire system.
     */
    public id: string | null = null

    /**
     * The id of the organization that owns this project.
     * Must be set by the caller before save — backend org enforcement rejects entities
     * with a missing or mismatched organizationId.
     */
    public organizationId!: string

    /**
     * The id of the application that this project belongs to.
     * All application ids are unique throughout the entire system.
     */
    public applicationId: string

    /**
     * The name of the project.
     */
    public name: string

    /**
     * The description of the project.
     */
    public description?: string

    /**
     * The source of truth for the project.
     */
    public sourceOfTruth: ProjectType | null = null

    /**
     * Full name ({@code owner/repo}) of the GitHub repository backing this project.
     * Stamped at create time by the server-side repo provisioner.
     */
    public repoFullName!: string

    /**
     * GitHub's stable repository id. Survives renames on the GitHub side.
     */
    public repoId!: number

    /**
     * Default branch of the backing repository at the time it was provisioned.
     */
    public defaultBranch!: string

    /**
     * Visibility chosen for the backing repository at create time. The SPA sets
     * this before save; the platform passes it through to GitHub.
     */
    public repoPrivate: boolean = true

    /**
     * The date and time the project was updated.
     */
    public updated: number | null = null

    constructor(id: string | null, applicationId: string, name: string, description?: string) {
        this.id = id
        this.applicationId = applicationId
        this.name = name
        this.description = description
    }
}
